parquet-column/src/main/java/org/apache/parquet/column/impl/ColumnReaderImpl.java

```diff
@@ -546,6 +573,11 @@ private void initDataReader(Encoding dataEncoding, byte[] bytes, int offset, int
     } catch (IOException e) {
       throw new ParquetDecodingException("could not read page in col " + path, e);
     }
+
+    if (dataEncoding.requiresSequentialReads(writerVersion) && previousReader != null) {
+      // previous reader can only be set if reading sequentially
+      ((RequiresPreviousReader) dataColumn).setPreviousReader(previousReader);
```

Comment:
We don't want this to be a normal part of the API, so using an interface like this to carry the method is a good practice. This also allows the reader to expose just the one method and not expose its internal state.